src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -73,13 +79,143 @@ internal static LanguageService LanguageServiceInstance
+        {
+            try
+            {
+                ScriptingListObjectsOperation operation = new ScriptingListObjectsOperation(parameters, requestContext);
+                RunListObjectsTask(operation);
+                await requestContext.SendResult(new ScriptingListObjectsResult { OperationId = operation.OperationId });
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Handles request to execute start the script operation.
+        /// </summary>
+        public async Task HandleScriptExecuteRequest(ScriptingParams parameters, RequestContext<ScriptingResult> requestContext)
+        {
+            try
+            {
+                ScriptingScriptOperation operation = new ScriptingScriptOperation(parameters, requestContext);
+                RunScriptTask(operation);
+                await requestContext.SendResult(new ScriptingResult { OperationId = operation.OperationId });
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Handles request to cancel the database script operation
+        /// </summary>
+        public async Task HandleScriptCancelRequest(ScriptingCancelParams parameters, RequestContext<ScriptingCancelResult> requestContext)
+        {
+            try
+            {
+                ScriptingOperation operation = null;
+                if (this.ActiveOperations.TryRemove(parameters.OperationId, out operation))
+                {
+                    operation.Cancel();
+                }
+
+                await requestContext.SendResult(new ScriptingCancelResult());
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Runs the async task to list scriptable objects.
+        /// </summary>
+        private void RunListObjectsTask(ScriptingListObjectsOperation operation)
+        {
+            Task.Run(async () =>
+            {
+                try
+                {
+                    this.ActiveOperations[operation.OperationId] = operation;
+                    await operation.Execute();
+                }
+                catch (Exception e)
+                {
+                    await operation.RequestContext.SendError(e);
+                }
+                finally
+                {
+                    ScriptingOperation temp;
+                    this.ActiveOperations.TryRemove(operation.OperationId, out temp);
+                }
+            });
+        }
+
+        /// <summary>
+        /// Runs the async task to script objects.
+        /// </summary>
+        private void RunScriptTask(ScriptingScriptOperation operation)
+        {
+            Task.Run(async () =>
```

Comment:
But I'm also curious since this pattern is seemingly reused several times, why isn't this method used more than once? Or more concisely, what makes this private method different than the other implementations of the same pattern.